docs/pages/docs/core-concepts/interfaces-and-unions.mdx

```diff
@@ -0,0 +1,312 @@
+    newsfeed: (): NewsfeedItem[] => {
+      // Implementation to fetch newsfeed items
+    }
+  }
+}
+```
+
+## Automatic Type Resolution
+
+One of the key features of Pylon's interface and union support is automatic type resolution. Unlike other GraphQL servers that require manual implementation of a `__resolveType` function, Pylon automatically resolves the correct type based on the shape of your data.
+
+This means you don't need to write code like this:
+
+```typescript
+// This is NOT needed in Pylon
+const resolveType = {
+  SearchResult: {
+    __resolveType(obj) {
+      if (obj.name) return 'User'
+      if (obj.title) return 'Post'
+      return null
+    }
+  }
+}
+```
+
+Pylon handles this automatically, simplifying your code and reducing the potential for errors.
+
+## Shared Properties
+
+When you define a union type where the member types have shared properties, Pylon automatically creates an interface for these shared properties.
+
+### Example
+
+```typescript
+type Bar = {
+  id: ID
+  title: string
+}
+
+type Foo = {
+  id: ID
+  name: string
+}
+
+type Example = Foo | Bar
+```
+
+This generates the following GraphQL schema:
+
+```graphql
+interface Example {
+  id: ID!
+}
+
+type Foo implements Example {
+  id: ID!
+  name: String!
+}
+
+type Bar implements Example {
+  id: ID!
+  title: String!
+}
+```
+
+This feature helps to maintain type safety and allows for more efficient querying of common properties.
+
+## Multiple Interfaces
+
+Pylon supports implementing multiple interfaces, mirroring the GraphQL specification. This allows for complex type hierarchies and flexible schema design.
+
+### Example
+
+```typescript
+interface Timestamped {
+  createdAt: Date
+  updatedAt: Date
+}
+
+interface Authored {
+  author: User
+}
+
+class BlogPost implements Node, Timestamped, Authored {
+  constructor(
+    public id: ID,
+    public title: string,
+    public createdAt: Date,
+    public updatedAt: Date,
+    public author: User
+  ) {}
+}
+```
+
+This will generate a GraphQL type that implements all three interfaces:
+
+```graphql
+type BlogPost implements Node & Timestamped & Authored {
+  id: ID!
+  title: String!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+  author: User!
+}
+```
+
+## Type Inference
+
+Pylon's type inference capabilities allow it to generate appropriate GraphQL schemas even from simple object structures. This means you can often define your types implicitly through your data structures.
+
+### Example
+
+```typescript
+const nodes = [
+  {
+    id: '1',
+    name: 'John Doe'
+  },
+  {
+    id: '4',
+    title: 'Hello, Pylon!'
+  }
+]
+
+export const graphql = {
+  Query: {
+    node: (id: ID) => {
+      return nodes.find(node => node.id === id)
+    }
+  }
+}
+```
+
+Pylon will automatically generate an interface and two types with generated names based on the structure of the `nodes` array. This can be particularly useful for rapid prototyping or when working with existing data structures.
+
```

Comment:
## Project Example

To give you a deeper insight, we have built an [example project](link-to-repo-directory) in which the new interface and union types are used.